src/Microsoft.AspNet.Razor/Generator/GeneratedTagHelperContext.cs

```diff
@@ -52,21 +58,53 @@ public GeneratedTagHelperContext()
         public string ScopeManagerEndMethodName { get; set; }
 
         /// <summary>
+        /// The name of the property used to determine if a tag helper output's content was set.
+        /// </summary>
+        public string OutputContentSetPropertyName { get; set; }
+
+        /// <summary>
         /// The name of the method used to generate a tag helper output's start tag.
         /// </summary>
         public string OutputGenerateStartTagMethodName { get; set; }
 
         /// <summary>
+        /// The name of the method used to generate a tag helper output's pre-content.
```

Comment:
nit: you have many more places where you're using "pre content" (no hyphen).  be consistent.